src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+
+        for (size_t i : {5, 6, 9, 10})
+            weights[i] = weight_vals[ColorIndex::GREEN];
+
+        Dispatcher::set_deltas(deltas, delta_vals, root);
+    }
+
+protected:
+    enum ColorIndex : uint8_t { GREEN = 0, ORANGE, RED };
+    constexpr static float delta_vals_[]  = {-1.5, -.5, .5, 1.5};
+    constexpr static float weight_vals_[] = {
+        0.140625, // 0 == GREEN
+        0.046875, // 1 == ORANGE
+        0.015625  // 2 == RED
+    };
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>>
+{
+    using Dispatcher = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<4>>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_val_, deltas, delta_val_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const& weight_val, //
+                      Deltas& deltas, float const& delta_val)
+    {
+        Dispatcher::set_weights(weights, weight_val);
+        Dispatcher::set_deltas(deltas, delta_val);
+    }
+
+protected:
+    constexpr static float delta_val_  = 0.683734;
+    constexpr static float weight_val_ = 0.25;
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>>
+    : public Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<4>, Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<8>, Color::PINK, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<8>>
+{
+    using Dispatcher = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Cross>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        Dispatcher::set_weights(weights, weight_vals);
+        Dispatcher::set_deltas(deltas, delta_vals);
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {0.828428, 1.236701};
+    constexpr static float weight_vals_[] = {0.178624, 0.071376};
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<8>>
+    : public Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<8>, Color::PINK,
+                      Color::PURPLE>
```

Comment:
ok, I might get it... this ctor is the one without reference to the color... so less arguments to use it. but as it make reference to the other one, we keep in mind the color pattern we invoke... is that it ?

Reply:
Yes, it's so we can have a configuration without specifying the color.
This is considered the default pattern when no color is specificed.
Which we need for python which does not know of colors.